src/ui/pages/Backup/RecoveryPhrase.tsx

```diff
@@ -11,35 +10,44 @@ import ArrowLeftIcon from 'jsx:src/ui/assets/arrow-left.svg';
 import { HStack } from 'src/ui/ui-kit/HStack';
 import { useCopyToClipboard } from 'src/ui/shared/useCopyToClipboard';
 import CopyIcon from 'jsx:src/ui/assets/copy.svg';
-import { BlurredToggle } from 'src/ui/pages/BackupWallet/BlurredToggle';
 import { UnstyledButton } from 'src/ui/ui-kit/UnstyledButton';
 import { CircleSpinner } from 'src/ui/ui-kit/CircleSpinner';
-import { clipboardWarning } from 'src/ui/pages/BackupWallet/clipboardWarning';
 import { SeedType } from 'src/shared/SeedType';
-import { useSizeStore } from '../useSizeStore';
-import * as helperStyles from '../shared/helperStyles.module.css';
-import { isSessionExpiredError } from '../shared/isSessionExpiredError';
-import { usePendingRecoveryPhrase } from '../shared/usePendingRecoveryPhrase';
+import { useWindowSizeStore } from 'src/ui/shared/useWindowSizeStore';
+import * as helperStyles from 'src/ui/features/onboarding/shared/helperStyles.module.css';
+import { isSessionExpiredError } from 'src/ui/shared/isSessionExpiredError';
+import { BlurredToggle } from 'src/ui/components/BlurredToggle';
+import { useGoBack } from 'src/ui/shared/navigation/useGoBack';
+import { useRecoveryPhrase } from './useRecoveryPhrase';
+import { clipboardWarning } from './clipboardWarning';
 
-export function Backup({
+export function RecoveryPhrase({
+  groupId,
   onNextStep,
   onSkip,
+  onSessionExpired,
 }: {
-  onNextStep(): void;
-  onSkip(): void;
+  groupId?: string;
+  onNextStep: () => void;
+  onSkip?: () => void;
+  onSessionExpired: () => void;
 }) {
-  const { isNarrowView } = useSizeStore();
-  const navigate = useNavigate();
-  const { data: mnemonic, error, isLoading } = usePendingRecoveryPhrase();
+  const { isNarrowView } = useWindowSizeStore();
+
+  const {
+    data: recoveryPhrase,
+    error,
+    isLoading,
+  } = useRecoveryPhrase({ groupId });
 
   useEffect(() => {
     if (isSessionExpiredError(error)) {
```

Comment:
let's not compute error when not necessary:
```suggestion
    if (isError && isSessionExpiredError(error)) {
```

Reply:
I didn't get it, because `isSessionExpiredError` is defined like this:
```
export function isSessionExpiredError(e: unknown) {
  if (!e) {
    return false;
  }
  return getError(e).message === 'Session expired';
}
```
So it just returns `false` in case `!e` evaluates to `false`

Reply:
Doesn't look right to rely on internal implementation. `isError` flag that comes from react-query is reliable
Also not sure it's good to handle falsy error there
Also one-letter param is not the best style

Reply:
Alright, I'll updated everywhere then, there are 10 places where we use `isSessionExpiredError`